README.md

```diff
@@ -1,77 +1,88 @@
-# Role Name
-![template](https://github.com/linux-system-roles/template/workflows/tox/badge.svg)
+# PostgreSQL system role
+![CI Testing](https://github.com/linux-system-roles/template/workflows/tox/badge.svg)
 
-A template for an ansible role which configures some GNU/Linux subsystem or
-service. A brief description of the role goes here.
+This role installs, configures, and starts PostgreSQL Server.
 
-## Requirements
-
-Any pre-requisites that may not be covered by Ansible itself or the role should
-be mentioned here. For instance, if the role uses the EC2 module, it may be a
-good idea to mention in this section that the `boto` package is required.
+The role also optimizes the datadase server settings to improve performance.
 
+The role currently works with PostgreSQL server 10 12 and 13.
 ## Role Variables
-
-A description of all input variables (i.e. variables that are defined in
-`defaults/main.yml`) for the role should go here as these form an API of the
-role.
-
-Variables that are not intended as input, like variables defined in
-`vars/main.yml`, variables that are read from other roles and/or the global
-scope (ie. hostvars, group vars, etc.) can be also mentioned here but keep in
-mind that as these are probably not part of the role API they may change during
-the lifetime.
-
-Example of setting the variables:
-
+### postgresql\_verison
+Allow set up version of Postgresql server. This role supports Postgresql 10 12 and 13
 ```yaml
-template_foo: "oof"
-template_bar: "baz"
+postgresql_version: "13"
+```
+### postgresql\_password
+Set up default password for database super user `postgres`
+```yaml
+postgresql_password: "mysecretpassword"
+```
+### pg\_hba\_conf
+A description of input variables that are not reqiured. Upstream configuration is used by default.
+Usage of `pg_hba_conf` causes replacement of default upstream configuration
+```yaml
+pg_hba_conf:
+  - type: local
+    database: all
+    user: all
+    auth_method: peer
+  - type: host
+    database: all
+    user: all
+    address: '127.0.0.1/32'
+    auth_method: ident
+  - type: host
+    database: all
+    user: all
+    address: '::1/128'
+    auth_method: ident
+```
+### postgresql\_server\_conf
+Usage of `postgresql_server_conf` adds defined values at the end of postgresql.conf.
+So the default ones are overwritten.
+```yaml
+postgresql_server_conf:
+  ssl: on
+  shared_buffers: 128 MB
+  huge_pages: try
+```
+### ssl\_enable
+To set up ssl connection it's necessary to set up `ssl_enable` variable and provide server certificate and key.
+```yaml
+ssl_enable: yes
+```
+### cert\_name
```

Comment:
add the postgresql_ prefix
It would be nice to support cert residing both locally and remotely. Like what `remote_src` var in the copy module does. So that customers have more flexibility.

Reply:
Right - see https://linux-system-roles.github.io/documentation/tls_crypto_parameter_and_key_names.html for our conventions about cert/key naming